src/actions/sources.js

```diff
@@ -101,8 +101,15 @@ export function newSources(sources: Source[]) {
       source => !getSource(getState(), source.id)
     );
 
+    dispatch({
+      type: "ADD_SOURCES",
+      sources
+    });
+
     for (const source of filteredSources) {
-      await dispatch(newSource(source));
+      await dispatch(loadSourceMap(source));
```

Comment:
maybe we can send the promises all at once using Promise.all? and if we move to sagas we can use forks